### PR TITLE
Release 3.7 subtensor integration

### PR DESCRIPTION
**Pull request by isabella618033**
- Based from finney, as well supporting backward compatibility with nakamoto.
- Bug fixes on both nakamoto cli and finney cli.
- Classes that was left keeping both nakamoto and finney implementation: cli, metagraph, subtensor'
    - Checked by subtensor.network or config.subtensor.network

### CODE REVIEW

File: bittensor/_neuron/text/core_server/__init__.py [hunk 1]
```diff
@@ -638,3 +629,27 @@ def time_check():
...
             return True
+
+    def get_neuron(self):
+        if self.subtensor.network == 'finney':
+            nn = self.subtensor.get_neuron_for_pubkey_and_subnet(self.wallet.hotkey.ss58_address, netuid = self.config.netuid)
+        elif self.subtensor.network == 'nakamoto':
+            nn = self.subtensor.neuron_for_pubkey(self.wallet.hotkey.ss58_address)
+        return nn
+
+    def get_n(self):
```
Comment (eduardogr) [inline bittensor/_neuron/text/core_server/__init__.py]: @isabella618033 what do you think about having more semantic in this function name??

From my point of view `get_n` should expose something more specific

File: bittensor/_neuron/text/core_validator/__init__.py [hunk 2]
```diff
@@ -553,22 +577,22 @@ def run_epoch( self ):
             weights=sample_weights.detach().to('cpu'),
             netuid = self.config.netuid,
             wallet=self.wallet,
-            version_key=bittensor.__version_as_int__, # TODO: correct?
+            version_key=1,
```
Comment (eduardogr) [inline bittensor/_neuron/text/core_validator/__init__.py:574]: what is this about?

File: bittensor/_neuron/text/core_validator/__init__.py [hunk 3]
```diff
@@ -126,6 +130,7 @@ def __init__(
             self.config.axon._mock = True
         print ( self.config )
 
+        copy.deepcopy(self.config)
```
Comment (eduardogr) [inline bittensor/_neuron/text/core_validator/__init__.py]: why are executing deepcopy three times? Do we need that? copy.deepcopy returns the result as a value right? So, why are not receiving the result?

**Review by eduardogr — COMMENTED** [review]
Nice Done @isabella618033 !

I'd approve but first i'd like to resolve some comment I left. After that everything LGTM. Nicely done!

File: bittensor/_neuron/text/core_validator/__init__.py (hunk 3 above)
Comment (isabella618033) [reply bittensor/_neuron/text/core_validator/__init__.py]: It was for some debugging purpose, and it is redundant now.

File: bittensor/_neuron/text/core_validator/__init__.py (hunk 2 above)
Comment (Eugene-hu) [reply bittensor/_neuron/text/core_validator/__init__.py:574]: The version key should be the bittensor version. Basically, the chain will check if the version key matches the latest bittensor version. However, that is not working right now, so it is set to 1

File: bittensor/_neuron/text/core_server/__init__.py (hunk 1 above)
Comment (isabella618033) [reply bittensor/_neuron/text/core_server/__init__.py]: Renamed get_n to get_neuron_num.

File: bittensor/_neuron/text/core_validator/__init__.py (hunk 2 above)
Comment (isabella618033) [reply bittensor/_neuron/text/core_validator/__init__.py:574]: Then are we keeping it as 1?

Comment (Eugene-hu) [reply bittensor/_neuron/text/core_validator/__init__.py:574]: For now, yes. I do not believe that feature is turned on yet.

Comment (isabella618033) [reply bittensor/_neuron/text/core_validator/__init__.py:574]: Okay! Just reverted it to 1:)

File: bittensor/_neuron/text/core_validator/__init__.py (hunk 3 above)
Comment (eduardogr) [reply bittensor/_neuron/text/core_validator/__init__.py]: can we remove it now??

**Review by Eugene-hu — APPROVED** [review]

File: bittensor/_neuron/text/core_validator/__init__.py [hunk 4]
```diff
@@ -48,6 +48,7 @@
 from loguru import logger
 from threading import Lock
 from prometheus_client import Counter, Gauge, Histogram, Summary, Info
+import copy
```
Comment (eduardogr) [inline bittensor/_neuron/text/core_validator/__init__.py]:
```suggestion
```

Comment (eduardogr) [reply bittensor/_neuron/text/core_validator/__init__.py]: since we are not using it anymore

**Review by eduardogr — APPROVED** [review]
LGTM Thanks @isabella618033  !

**Pull request merged**